### PR TITLE
[videoplayer] Some cleanup and fixes

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

**Comment by ace20022** [comment]
@FernetMenta for review. btw I'm stuck with the enable stream stuff because it breaks dvds. I'm wondering how that dvdnav/selectionstream stuff could work that long.

**Comment by FernetMenta** [comment]
@ace20022 I would start with this: https://github.com/FernetMenta/xbmc/commit/16b20eb22b8aa4d2b6cce26aebdb3a18d43a532c

what is broken with DVDs?

**Comment by Voyager1** [comment]
guessing at the problems that @ace20022 is referring to, I'd say it's the whole xbmc vs. external stream id's conversion, see CDVDInputStreamNavigator::Convert{Audio/Subtitle}StreamId_{ExternalToXBMC/XBMCToExternal}

**Comment by ace20022** [comment]
> what is broken with DVDs?

What @Voyager1 said is one thing.
Switching streams on dvds work with CheckBetterStream. Videoplayer waits till a packet with the correct dvdnav id comes it's way. Initialize streams as skippable does not work in this case, since enablingstream uses the selectionstream id which is not identical to the unique id. 
Using the dvdnavId might work, but enableStream must be made aware of that. Furthermore dvd menus do not work (maybe for the same reason).

@FernetMenta is this pr okay?

jenkins build this please

**Comment by FernetMenta** [comment]
@ace20022 +1

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -3250,8 +3250,21 @@ void CVideoPlayer::UpdateStreamInfos()
...
+        {
+          SelectionStream& select = m_SelectionStreams.Get(STREAM_VIDEO, i);
+          select.width = ((CDemuxStreamVideo*)stream)->iWidth;
+          select.height = ((CDemuxStreamVideo*)stream)->iHeight;
+        }
+      }
+      else
+      {
+        s.width = ((CDemuxStreamVideo*)stream)->iWidth;
+        s.height = ((CDemuxStreamVideo*)stream)->iHeight;
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: static_cast please (I know it wasn't in the original)

**Comment by ace20022** [comment]
@garbear thanks!

jenkins build this please

**Comment by ace20022** [comment]
jenkins build this please

**Pull request merged**